common/frame_manager.go

```diff
@@ -627,10 +629,16 @@ func (m *FrameManager) NavigateFrame(frame *Frame, url string, opts goja.Value)
 		if err != nil {
 			k6common.Throw(rt, err)
 		}
+
 		event = data.(*NavigationEvent)
 		if event.newDocument.documentID != newDocumentID {
-			k6common.Throw(rt, errors.New("navigation interrupted by another one"))
-		} else if event.err != nil {
+			m.logger.Debugf("FrameManager:NavigateFrame:interrupted",
+				"fmid:%d fid:%v furl:%s url:%s docID:%s newDocID:%s",
+				fmid, fid, furl, url, event.newDocument.documentID, newDocumentID)
+		} else if event.err != nil &&
+			// TODO: A more graceful way of avoiding Throw()?
```

Comment:
@imiric What do you mean here?

Reply:
I mean that this stringy check is clunky, but I didn't look too much into it. E.g. could we avoid setting `event.err` somewhere lower level so that this is never reached?

Reply:
Yeah, possibly. I didn't spend much time on it, just wanted to avoid calling `Throw()` in this case, but yeah, it needs more investigation, hence the `TODO`. :)